packages/api-specification/transform-type-info.js

```diff
@@ -0,0 +1,106 @@
+const jsont = require('json-transforms');
+const fs = require('fs');
+const program = require('commander');
+const json2html = require('html2json').json2html;
+const matter = require('gray-matter');
+
+program
+  .option('-i, --infile [filename]', 'Input type definitions file', 'type-info.json')
+  .option('-o, --outfile [filename]', 'Transformed output')
+  .parse(process.argv);
+
+if (!program.outfile) {
+  console.error('You must specify an output file');
+  process.exit(0);
+}
+
+const typeInfo = JSON.parse(fs.readFileSync(program.infile));
+
+const flatten = (arr) => arr.reduce((a, b) => a.concat(b), []);
+
+// html elements are a bit complex, this simplifies their construction
+const element = (tag, child) => ({
+  node: 'element',
+  tag,
+  // if child is not an array, turn it into one - also, filter out any undefined children
+  child: (Array.isArray(child) ? child : [child]).filter(d => d)
+});
+
+// text elements have not children
+const text = (text) => ({
+  node: 'text',
+  text
+});
+
+// there are a number of elements that just host a single text node, this
+// simplifies their construction
+const textElement = (tag) => (body) => element(tag, text(body));
+const h4 = textElement('h4');
+const h1 = textElement('h1');
+const h2 = textElement('h2');
+const p = textElement('p');
+const dt = textElement('dt');
+const dd = textElement('dd');
+
+// there are a number of elements that host multiple children, this
+// simplifies their construction
+const parentElement = (tag) => (children) => element(tag, children);
+const section = parentElement('section');
+const dl = parentElement('dl');
+
+const rules = [
+  // perform a 'deep' match to find any class declarations, then recursively
+  // matches from this point onwards
+  jsont.pathRule(
+    '..children{.kindString === "Class"}', d => ({
+      node: 'root',
+      child: d.runner()
+    })
+  ),
+  jsont.pathRule(
+    '.{.kindString === "Class"}', d =>
+      section([
+        h1(d.match.name),
+        section(d.runner(d.match.children))
+      ])
+  ),
+  jsont.pathRule(
+    '.{.kindString === "Method"}', d =>
+      // we take just the first signature as the definition of the method, not
+      // sure when there might be multiple signatures?!
```

Comment:
I wasn't sure either. It seems like Typescript supports multiple method signatures, but only 1 method implementation. E.g.
```javascript
aMethod(a: string);
aMethod(a: number);
aMethod(a: any) {
  console.log(a);
}
```
So it seems like a very specialized use case. The same could be achieved with union types I think